fluvii/kafka_tools/fluvii_toolbox.py

```diff
@@ -92,51 +93,107 @@ def create_topics(self, topic_config_dict, ignore_existing_topics=True):
                 futures[topic].result()
         LOGGER.info(f'Created topics: {list(topic_config_dict.keys())}')
 
-    def alter_topics(self, topic_config_dict, retain_configs=True, ignore_missing_topics=True, protected_configs=None):
+    def alter_topics(self, topic_config_dict, retain_configs=True, ignore_missing_topics=True, ignore_field_list=None):
         """
-        {'topic_a': {'partitions': 1, 'replication.factor': 1, 'segment.ms': 10000}, 'topic_b': {etc}}
+        {'topic_a': {'segment.ms': 10000}, 'topic_b': {etc}}
+        NOTE: you cannot change partitions or replication factor this way
+        NOTE: "retain_configs" means keeping a previous value if you did not provide a new value for a given setting.
+        TODO: make a preview option
         """
         self._set_admin()
-        if not protected_configs:
-            protected_configs = []
+        if not ignore_field_list:
```

Comment:
The 'ignore_field_list' parameter has a default value of None, but it is assigned a value of an empty list in the function body. It would be more appropriate to define the default value of this parameter as an empty set (ignore_field_list=None), so that it can be properly handled when passed as an argument.

Reply:
I think it's fair to resolve this based on our discussion about argument types handed to a function? Feel free to unresolve otherwise.